packages/react-button/src/components/Button/useButtonStyles.ts

```diff
@@ -13,587 +12,278 @@ export const buttonSpacing = {
+    lineHeight: theme.global.type.lineHeights.base[300],
+  }),
+  childrenSmall: theme => ({
+    fontSize: theme.global.type.fontSizes.base[200],
+    fontWeight: theme.global.type.fontWeights.regular,
+    lineHeight: theme.global.type.lineHeights.base[200],
+  }),
+  childrenLarge: theme => ({
+    fontSize: theme.global.type.fontSizes.base[400],
+    lineHeight: theme.global.type.lineHeights.base[400],
+  }),
+  icon: {
+    alignItems: 'center',
+    display: 'inline-flex',
+    justifyContent: 'center',
+
+    fontSize: '20px',
+    height: '20px',
+    width: '20px',
   },
-  iconLarge: theme => {
-    const buttonTokens = makeButtonTokens(theme);
-
-    return {
-      fontSize: buttonTokens.large.iconFontSize,
-      height: buttonTokens.large.iconHeight,
-      width: buttonTokens.large.iconWidth,
-    };
+  iconLarge: {
+    fontSize: '24px',
+    height: '24px',
+    width: '24px',
   },
 });
 
-export const useButtonStyles = (state: ButtonState, selectors: ButtonStyleSelectors) => {
+export const useButtonStyles = (state: ButtonState) => {
   const styles = useStyles();
   state.className = mergeClasses(
     styles.root,
-    selectors.iconOnly && styles.rootIconOnly,
-    selectors.primary && styles.rootPrimary,
-    selectors.subtle && styles.rootSubtle,
-    selectors.transparent && styles.rootTransparent,
-    selectors.disabled && styles.rootDisabled,
-    selectors.disabled && selectors.primary && styles.rootDisabledPrimary,
-    selectors.disabled && selectors.subtle && styles.rootDisabledSubtle,
-    selectors.disabled && selectors.transparent && styles.rootDisabledTransparent,
-    selectors.size === 'small' && styles.rootSmall,
-    selectors.size === 'large' && styles.rootLarge,
-    selectors.iconOnly && selectors.size === 'small' && styles.rootIconOnlySmall,
-    selectors.iconOnly && selectors.size === 'large' && styles.rootIconOnlyLarge,
+    state.iconOnly && styles.rootIconOnly,
+    state.primary && styles.rootPrimary,
+    state.subtle && styles.rootSubtle,
+    state.transparent && styles.rootTransparent,
+    state.disabled && styles.rootDisabled,
+    state.disabled && state.primary && styles.rootDisabledPrimary,
+    state.disabled && state.subtle && styles.rootDisabledSubtle,
+    state.disabled && state.transparent && styles.rootDisabledTransparent,
+    state.size === 'small' && styles.rootSmall,
+    state.size === 'large' && styles.rootLarge,
+    state.iconOnly && state.size === 'small' && styles.rootIconOnlySmall,
+    state.iconOnly && state.size === 'large' && styles.rootIconOnlyLarge,
     state.className,
   );
 
   if (state.children) {
     state.children.className = mergeClasses(
       styles.children,
-      selectors.size === 'small' && styles.childrenSmall,
-      selectors.size === 'large' && styles.childrenLarge,
+      state.size === 'small' && styles.childrenSmall,
+      state.size === 'large' && styles.childrenLarge,
       state.children.className,
     );
   }
```

Comment:
I know this isn't new in this PR, so it doesn't need to be addressed now... I don't think it's safe to assume that children is a DOM node that can take a className. It could be an array of children, a fragment, or plain text. Or is there something I missed that ensures that button has a single DOM node as a child?

Reply:
You are right about that, although I'll address this in a separate PR. I've made this issue to track it #18471.